### PR TITLE
Add possibility to add multiple phpstan configs

### LINKED ISSUE #8141: Add possibility of adding multiple phpstan neon files
# Feature Request

As mention in https://github.com/rectorphp/rector/issues/8140 it would be great to configure the phpstan extension from the rector.php file and not have additonal phpstan or change the own phpstan configuration for it.
 
## Diff 

```diff
-    $rectorConfig->phpstanConfig(__DIR__ . '/phpstan.neon');
+    $rectorConfig->phpstanConfigs([
+        '__DIR__ . '/phpstan.neon',
+        '__DIR__ . '/vendor/phpstan/phpstan-doctrine/extension.neon',
+        '__DIR__ . '/vendor/phpstan/phpstan-symfony/extension.neon',
+    ]);
```

I think it would make sense then deprecate `phpstanConfig` in favor of `phpstanConfigs`. In cases of `rector-symfony`, `rector-doctrine`, ... the docs should be mentioning configuring this. So rector works correctly on symfony services and doctrine entities, ...

### PR DESCRIPTION
**Pull request by alexander-schranz**
Closes https://github.com/rectorphp/rector/issues/8141

### CODE REVIEW

**Comment by samsonasik** [comment]
would be great to have both available imo, that's like `$rectorConfig->rule()` and `$rectorConfig->rules()`

**Comment by alexander-schranz** [comment]
@samsonasik Okay then we need to discuss the defaults of the following:

**Case A**: Using `phpstanConfigs` without `phpstanConfig` e.g.:

```php
$rectorConfig->phpstanConfigs(['some-test.neon'])
```

Should the current default `phpstan.neon` still be loaded or not?

**Case B**: Using `phpstanConfigs` and `phpstanConfig`:

```php
$rectorConfig->phpstanConfig('some-test.neon');
$rectorConfig->phpstanConfigs(['some-test.neon']);
```

Does `phpstanConfigs` overwrite `phpstanConfig` or not?

**Case C**: Using `phpstanConfigs` before `phpstanConfig`:

```php
$rectorConfig->phpstanConfigs(['some-test.neon']);
$rectorConfig->phpstanConfig('some-test.neon');
```

Does `phpstanConfig` overwrite `phpstanConfigs` or not?

**Case D**: Multiple `phpstanConfigs` calls:

```php
$rectorConfig->phpstanConfigs(['some-test.neon']);
$rectorConfig->phpstanConfigs(['some-other.neon']);
```

Does it replace or not load both?

**Case E**: Multiple `phpstanConfig` calls:

```php
$rectorConfig->phpstanConfig('some-test.neon');
$rectorConfig->phpstanConfig('some-other.neon');
```

Does it replace or not load both?

**Comment by samsonasik** [comment]
I think the config should merge like when we define:

```php
$rectorConfig->rule(FirstRector::class);

$rectorConfig->rules([
        SecondRector::class,
        ThirdRector::class,
]);
```

The result of rules are 3.

**Comment by alexander-schranz** [comment]
@samsonasik what I'm currently not understand how currently the fallback to `phpstan.neon` works if there is nothing configured.

**Comment by alexander-schranz** [comment]
Or is this comment here: https://github.com/rectorphp/rector-src/blob/e7f551069cd3efc29ccd7089f432f22556f52b0c/packages/Config/RectorConfig.php#L137 not longer true?

**Comment by samsonasik** [comment]
That seems no longer true afaik

File: build/config/config-downgrade.php [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
     $rectorConfig->skip(DowngradeRectorConfig::DEPENDENCY_EXCLUDE_PATHS);
 
-    $rectorConfig->phpstanConfig(__DIR__ . '/phpstan-for-downgrade.neon');
+    $rectorConfig->phpstanConfigs([__DIR__ . '/phpstan-for-downgrade.neon']);
```
Comment (samsonasik) [inline build/config/config-downgrade.php]: this can be reverted to use `phpstanConfig()` since only 1.

File: rector.php [hunk 2]
```diff
@@ -89,5 +89,5 @@
         RemovePhpVersionIdCheckRector::class => [__DIR__ . '/src/Util/FileHasher.php'],
     ]);
 
-    $rectorConfig->phpstanConfig(__DIR__ . '/phpstan-for-rector.neon');
+    $rectorConfig->phpstanConfigs([__DIR__ . '/phpstan-for-rector.neon']);
```
Comment (samsonasik) [inline rector.php]: this can be reverted to use `phpstanConfig()` since only 1.

**Review by samsonasik — APPROVED** [review]
Looks good to me 👍

**Review by staabm — COMMENTED** [review]
Could it be covered by a test?

**Comment by TomasVotruba** [comment]
@alexander-schranz Thank you :+1:

**Comment by bloep** [comment]
@TomasVotruba do you see any chance that we can get a Rector release with this fix in the near future?
That would solve a pain point for us 🙏

**Comment by TomasVotruba** [comment]
@bloep Sure, I'm on it :)

**Comment by TomasVotruba** [comment]
~~There you https://github.com/rectorphp/rector/releases/tag/0.17.14 :muscle:~~

**Comment by TomasVotruba** [comment]
I have just realized the container switch might be a BC break for anyone using Symfony internals (though they should not) :facepalm: 

So I changed replaced the tag to a minor version jump, just to be sure:
**https://github.com/rectorphp/rector/releases/tag/0.18.0**

**Use the Rector 0.18.0 tag instead**. My appologies for confusion  :pray:

**Pull request merged**